File: evento-app/evento-app/src/app/sistema/evento/Local.ts
import { Evento } from "./Evento";

export class Local{
    id!:number;
    estado: string;
    cidade: string;
    bairro: string;
    rua: string;
    nomeLocal: string;
    numero: number;
    eventoId: number;

    constructor(    estado: string,
        cidade: string,
        bairro: string,
        rua: string,
        nomeLocal: string,
        numero: number,
        eventoId: number){
            this.estado = estado;
            this.cidade = cidade;
            this.bairro = bairro;
            this.rua = rua;
            this.nomeLocal = nomeLocal;
            this.numero = numero;
            this.eventoId = eventoId
        }
}

export class LocalDTO{
    id!:number;
    estado: string;
    cidade: string;
    bairro: string;
    rua: string;
    nomeLocal: string;
    numero: number;
    evento: Evento;

    constructor(    estado: string,
        cidade: string,
        bairro: string,
        rua: string,
        nomeLocal: string,
        numero: number,
        evento: Evento){
            this.estado = estado;
            this.cidade = cidade;
            this.bairro = bairro;
            this.rua = rua;
            this.nomeLocal = nomeLocal;
            this.numero = numero;
            this.evento = evento;
        }
}

File: evento/src/main/java/com/myevent/eventos/entity/Local.java
package com.myevent.eventos.entity;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Local {

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	private String estado;
	
	private String Cidade;
	
	private String bairro;
	
	private String rua;
	
	private String nomeLocal;
	
	private Integer numero;
	
    @ManyToOne
    @JoinColumn(name = "evento_id")
    private Evento evento;
	
	private Boolean status;
}
